### PR TITLE
docs: stats guide for setting static to dynamic

### PR DESCRIPTION
**Pull request by chrysan**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v6.0 (TiDB 6.0 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -1442,54 +1501,59 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
...
-+---------+------------+----------------+---------------------+--------------+-----------+
-| Db_name | Table_name | Partition_name | Update_time         | Modify_count | Row_count |
-+---------+------------+----------------+---------------------+--------------+-----------+
-| test    | t          | global         | 2022-05-27 20:50:53 |            0 |         5 |
-| test    | t          | p0             | 2022-05-27 20:23:34 |            1 |         2 |
-| test    | t          | p1             | 2022-05-27 20:50:52 |            0 |         2 |
-| test    | t          | p2             | 2022-05-27 20:50:08 |            0 |         2 |
-+---------+------------+----------------+---------------------+--------------+-----------+
-4 rows in set (0.00 sec)
+mysql> SET tidb_partition_prune_mode = dynamic;
```
[the remark this answers is not in the document]
Comment (chrysan) [reply partitioned-table.md]: 统一了，还是按照 session 操作，globalStats 准备好以后再切全局的 dynamic

**Comment by songrijie** [comment]
LGTM

File: partitioned-table.md [hunk 2]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+| test    | t          | global         | 2022-05-27 20:50:53 |            0 |         5 |
+| test    | t          | p0             | 2022-05-27 20:23:34 |            1 |         2 |
+| test    | t          | p1             | 2022-05-27 20:50:52 |            0 |         2 |
+| test    | t          | p2             | 2022-05-27 20:50:08 |            0 |         2 |
++---------+------------+----------------+---------------------+--------------+-----------+
+4 rows in set (0.00 sec)
+```
+
+若 analyze 过程中提示如下的 warning，说明分区的统计信息之间存在不一致，则需要重新收集分区或整个表统计信息使一致。
+```
```
Comment (shichun-0415) [inline partitioned-table.md:1364]:
```suggestion

```
```

File: partitioned-table.md [hunk 3]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+| Warning | 8244 | Build table: `t` column: `a` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions      
+```
+也可以使用脚本来统一更新所有的分区表统计信息，详见 [为动态裁剪模式更新所有分区表统计信息](/partitioned-table.md#为动态裁剪模式更新所有分区表统计信息)
```
Comment (shichun-0415) [inline partitioned-table.md:1367]:
```suggestion

也可以使用脚本来统一更新所有的分区表统计信息，详见 [为动态裁剪模式更新所有分区表统计信息](/partitioned-table.md#为动态裁剪模式更新所有分区表统计信息)。
```

File: partitioned-table.md [hunk 4]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+
+表级别统计信息准备好后，即可开启全局的动态裁剪模式。
+
+
```
Comment (shichun-0415) [inline partitioned-table.md:1371]:
```suggestion

```

File: partitioned-table.md [hunk 5]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+{{< copyable "sql" >}}
+
+```sql
+set global tidb_partition_prune_mode = dynamic
+```
+
+
```
Comment (shichun-0415) [inline partitioned-table.md:1378]:
```suggestion

```

File: partitioned-table.md [hunk 6]
```diff
@@ -1440,3 +1506,61 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
 从示例二结果可知，开启 `dynamic` 模式后，带 IndexJoin 的计划在执行查询时被选上。
 
 目前，静态和动态裁剪模式都不支持执行计划缓存。
+
+
```
Comment (shichun-0415) [inline partitioned-table.md:1510]:
```suggestion

```

File: partitioned-table.md [hunk 7]
```diff
@@ -1440,3 +1506,61 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
...
+执行批量更新：
+
+{{< copyable "sql" >}}
+
+```sql
+mysql> SET session tidb_partition_prune_mode = dynamic;
+mysql> source gatherGlobalStats.sql 
+```
+
+
```
Comment (shichun-0415) [inline partitioned-table.md:1566]:
```suggestion
```

File: partitioned-table.md [hunk 8]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
 set @@session.tidb_partition_prune_mode = 'dynamic'
 ```
 
+动态裁剪模式下的分区表需要用到表级别的汇总统计信息，即 GlobalStats，详见[动态裁剪模式下的分区表统计信息](/statistics.md#动态裁剪模式下的分区表统计信息)。
```
Comment (shichun-0415) [inline partitioned-table.md:1313]:
```suggestion
动态裁剪模式下，分区表需要用到表级别的汇总统计信息，即 GlobalStats。详见[动态裁剪模式下的分区表统计信息](/statistics.md#动态裁剪模式下的分区表统计信息)。
```

File: partitioned-table.md [hunk 9]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+
+从 `static` 静态裁剪模式切到 `dynamic` 动态裁剪模式时，需要手动检查和收集汇总统计信息。在刚切到 `dynamic` 的时候，分区表上仍然只有分区的统计信息，需要等到下一次 `auto-analyze` 周期才会更新生成汇总统计信息。
```
Comment (shichun-0415) [inline partitioned-table.md:1315]:
```suggestion
从 `static` 静态裁剪模式切到 `dynamic` 动态裁剪模式时，需要手动检查和收集统计信息。在刚切换到 `dynamic` 时，分区表上仍然只有分区的统计信息，需要等到下一次 `auto-analyze` 周期才会更新生成汇总统计信息。
```

File: partitioned-table.md [hunk 10]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
++---------+------------+----------------+---------------------+--------------+-----------+
+| test    | t          | global         | 2022-05-27 20:50:53 |            0 |         5 |
+| test    | t          | p0             | 2022-05-27 20:23:34 |            1 |         2 |
+| test    | t          | p1             | 2022-05-27 20:50:52 |            0 |         2 |
+| test    | t          | p2             | 2022-05-27 20:50:08 |            0 |         2 |
++---------+------------+----------------+---------------------+--------------+-----------+
+4 rows in set (0.00 sec)
+```
+
+若 analyze 过程中提示如下的 warning，说明分区的统计信息之间存在不一致，则需要重新收集分区或整个表统计信息使一致。
```
Comment (shichun-0415) [inline partitioned-table.md:1363]:
```suggestion
若 analyze 过程中提示如下 warning，说明分区的统计信息之间存在不一致，需要重新收集分区或整个表统计信息。
```

File: partitioned-table.md [hunk 11]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+
+```
+| Warning | 8244 | Build table: `t` column: `a` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions
+```
+
+也可以使用脚本来统一更新所有的分区表统计信息，详见 [为动态裁剪模式更新所有分区表统计信息](/partitioned-table.md#为动态裁剪模式更新所有分区表统计信息)。
```
Comment (shichun-0415) [inline partitioned-table.md:1369]:
```suggestion
也可以使用脚本来统一更新所有的分区表统计信息，详见[为动态裁剪模式更新所有分区表统计信息](/partitioned-table.md#为动态裁剪模式更新所有分区表统计信息)。
```

File: partitioned-table.md [hunk 12]
```diff
@@ -1440,3 +1506,58 @@ mysql> explain select /*+ TIDB_INLJ(t1, t2) */ t1.* from t1, t2 where t2.code =
 从示例二结果可知，开启 `dynamic` 模式后，带 IndexJoin 的计划在执行查询时被选上。
 
 目前，静态和动态裁剪模式都不支持执行计划缓存。
+
+#### 为动态裁剪模式更新所有分区表统计信息
```
Comment (shichun-0415) [inline partitioned-table.md:1510]: 下面的步骤建议添加编号 1、2、3、4，看着清晰一些。

Comment (chrysan) [reply partitioned-table.md:1510]: Done

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: partitioned-table.md (hunk 12 above)
Comment (qiancai) [inline partitioned-table.md:1510]:
```suggestion
#### 为动态裁剪模式更新所有分区表的统计信息
```

File: partitioned-table.md [hunk 13]
```diff
@@ -1310,6 +1310,72 @@ TiDB 访问分区表有两种模式，`dynamic` 和 `static`，目前默认使
...
+4 rows in set (0.00 sec)
+```
+
+若 analyze 过程中提示如下 warning，说明分区的统计信息之间存在不一致，需要重新收集分区或整个表统计信息。
+
+```
+| Warning | 8244 | Build table: `t` column: `a` global-level stats failed due to missing partition-level column stats, please run analyze table to refresh columns of all partitions
+```
+
+也可以使用脚本来统一更新所有的分区表统计信息，详见[为动态裁剪模式更新所有分区表统计信息](/partitioned-table.md#为动态裁剪模式更新所有分区表统计信息)。
```
Comment (qiancai) [inline partitioned-table.md:1369]:
```suggestion
也可以使用脚本来统一更新所有的分区表统计信息，详见[为动态裁剪模式更新所有分区表的统计信息](/partitioned-table.md#为动态裁剪模式更新所有分区表的统计信息)。
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**